corehq/apps/hqwebapp/tasks.py

```diff
@@ -45,7 +45,8 @@ def mark_subevent_gateway_error(messaging_event_id, error, retrying=False):
 @task(serializer='pickle', queue="email_queue",
       bind=True, default_retry_delay=15 * 60, max_retries=10, acks_late=True)
 def send_mail_async(self, subject, message, recipient_list, from_email=settings.DEFAULT_FROM_EMAIL,
-                    messaging_event_id=None, domain: str = None, use_domain_gateway=False):
+                    filename: str = None, content=None, messaging_event_id=None, domain: str = None,
```

Comment:
For the translation stuff specifically, (`file_obj.read()`) is already in bytes so, correct me if i'm wrong, I don't think it would need to be serialized by celery. 

But I think it's safer to keep it synchronous since the higher level function is already asynchronous so having email also asynchronous doesn't get us much - right? [36f9a52](https://github.com/dimagi/commcare-hq/pull/33692/commits/36f9a52214534f18718697a9892b9f63529c37a9)

Agreed on the interface consolidation opportunity you mentioned here and in the other comment. I'll open up another PR for that

Reply:
> I don't think it would need to be serialized by celery
That sounds plausible - I was thinking more about the storage issue.  I remember some thing a while back where we passed a user object as an arg to celery and it was big enough to cause issues, and we decided best practices would be to pass in object IDs and re-fetch the object in the task.  I don't remember the specifics though - I'll pose the question on slack.

I do also agree that leaving it synchronous makes sense, since we took all the time to generate the report in the task anyways.  One thing we do lose though is the retry functionality in the `send_mail_async` task, though I have no sense how useful that is in practice.

Reply:
All that makes sense.

Here's where I changed to use the same `file_attachments` interface
https://github.com/dimagi/commcare-hq/pull/33732/commits/a0873ce5d24d33db232e029a303f08d480e99f92